app/src/main/java/ch/epfl/cs311/wanderwave/ui/components/login/LoginScreenHeader.kt

```diff
@@ -2,34 +2,69 @@ package ch.epfl.cs311.wanderwave.ui.components.login
 import androidx.compose.material3.Icon
 import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import ch.epfl.cs311.wanderwave.R
+import ch.epfl.cs311.wanderwave.navigation.NavigationActions
+import ch.epfl.cs311.wanderwave.navigation.Route
+import ch.epfl.cs311.wanderwave.ui.components.utils.lerp
+import ch.epfl.cs311.wanderwave.ui.theme.placeholderColor
 
 @Composable
-fun LoginScreenHeader(modifier: Modifier) {
-
-  Column(
-      horizontalAlignment = Alignment.CenterHorizontally,
-      verticalArrangement = Arrangement.Center,
-      modifier = modifier.fillMaxWidth().padding(16.dp)) {
-        Icon(
-            painter = painterResource(id = R.drawable.app_logo),
-            contentDescription = "app logo",
-            tint = MaterialTheme.colorScheme.primary,
-            modifier =
-                Modifier.height(50.dp)
-                    .fillMaxWidth()
-                    .padding(start = 50.dp, end = 50.dp)
-                    .testTag("appLogo"))
+fun LoginScreenHeader(navigationActions: NavigationActions, modifier: Modifier) {
+  val startColor = MaterialTheme.colorScheme.primary
+  val endColor = placeholderColor
+  val colorSpots = List(10) { i -> lerp(startColor, endColor, i / 9f) }
+  FloatingActionButton(
+      onClick = { navigationActions.navigateTo(Route.ABOUT) },
+      containerColor = MaterialTheme.colorScheme.background,
+      modifier = Modifier.testTag("appIcon").padding(top = 30.dp)) {
+        Column(modifier = modifier.padding(start = 60.dp, end = 60.dp)) {
+          Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.End) {
+            Icon(
+                painter = painterResource(id = R.drawable.info),
+                contentDescription = "Info Icon",
+                tint = MaterialTheme.colorScheme.onSurface,
+                modifier = Modifier.size(30.dp))
+          }
+          Row(
+              horizontalArrangement = Arrangement.Center,
+              verticalAlignment = Alignment.CenterVertically,
+              modifier = Modifier.fillMaxWidth()) {
+                Icon(
+                    painter = painterResource(id = R.drawable.wanderwave_icon),
+                    contentDescription = "app icon",
+                    tint = colorSpots[0],
+                    modifier = Modifier.height(40.dp).testTag("appIcon"))
+                val anderwave: List<String> = listOf("a", "n", "d", "e", "r", "w", "a", "v", "e")
+                anderwave.forEachIndexed { index, letter ->
+                  FillLogo(letter = letter, colorSpot = index, colorSpots = colorSpots)
+                }
```

Comment:
If that's how we draw the logo, we should probably extract it to a reusable component. It would probably be easiest to just use something like an SVG instead.

Reply:
Agree. The way it's handled now is a bit clumsy